.circleci/config.yml

```diff
@@ -351,12 +351,29 @@ defaults:
 
   - steps_install_dependencies_osx: &steps_install_dependencies_osx
       steps:
-        # FIXME: We used to cache dependencies on macOS but now it takes longer than just installing
-        # them each time. See https://github.com/ethereum/solidity/issues/12925.
+        - restore_cache:
+            keys:
+              - dependencies-osx-{{ arch }}-{{ checksum ".circleci/osx_install_dependencies.sh" }}
+        - attach_workspace:
+            at: .
```

Comment:
You should now remove `attach_workspace` from jobs that use this step. Otherwise we're doing it twice. Which may or may not be safe to do.